avssync/metrics.go

```diff
@@ -35,12 +36,32 @@ var (
 	}, []string{"quorum"})
 )
 
-func StartMetricsServer(metricsAddr string) {
+type Metrics struct {
+	registry *prometheus.Registry
+	addr     string
+	logger   logging.Logger
+}
+
+func NewMetrics(addr string, logger logging.Logger) *Metrics {
 	registry := prometheus.NewRegistry()
 	registry.MustRegister(updateStakeAttempt, txRevertedTotal, operatorsUpdated)
```

Comment:
Talked with @dmanc about this.
Feel like we're not addressing the proper separation of concern (absolutely my fault for this, just realized now). Created https://github.com/Layr-Labs/eigensdk-go/issues/238 to do this in sdk as well.

But how are you guys registering the above metrics (operatorsUpdated, etc) with a separate registry?
I feel like ideal scenario would be that NewMetrics takes a registry as input, and then registers the variables against that registry. Maybe a hacky way that doesn't require too mcuh refactor is just to add a one-off function in this file `registerPromVariables(reg prometheus.Registry)` that just registers against some prom registry?